File: rails/callback.rb
def callback
  if params[:confirmations] && params[:order]
    order_id = params[:order]
    data = params
    invoice = data[:invoice]

    if data[:confirmations] >= data[:maxConfirmations]
      amount_paid = data[:inTransaction][:amount] / 10**data[:inTransaction][:exp]
      # compare $amountPaid with order total
      # compare $invoice with one saved in the database to ensure callback is legitimate
      # mark the order as paid
      render plain: invoice # stop further callbacks
    else
      # save number of confirmations to DB
      render plain: 'waiting for confirmations'
    end
  else
    render plain: 'error'
  end
end

File: nodejs/callback.js
const app = new (require('express').Router)()

app.post('/paybear/callback/:order', (req, res) => {
  if (req.body && req.params.order) {
    // const orderId = req.params.order
    const data = req.body
    const invoice = data.invoice

    // save data.confirmations - number of confirmations to DB

    if (data.confirmations >= data.maxConfirmations) {
      // const amountPaid = data.inTransaction.amount / Math.pow(10, data.inTransaction.exp)

      // compare $amountPaid with order total
      // compare $invoice with one saved in the database to ensure callback is legitimate
      // mark the order as paid
      res.send(invoice) // stop further callbacks
    } else {
      res.send('waiting for confirmations')
    }
  } else {
    res.send('error')
  }
})

module.exports = app
